pytng/pytng.pyx

```diff
@@ -1016,335 +1133,328 @@ cdef class TNGFileIterator:
-        """Gets the frame data off disk and into C level arrays"""
+        if datatype == TNG_FLOAT_DATA:
+            if dtype != np.float32:
+                raise TypeError(
+                    "PYTNG ERROR: dtype of array {} does not match TNG dtype float".format(dtype))
+            _float_view = <np.float32_t[:n_vals] > (< float*> values)
+            data[:, :] = np.asarray(_float_view, dtype=np.float32).reshape(
+                n_atoms, n_values_per_frame)
+
+        elif datatype == TNG_INT_DATA:
+            if dtype != np.int64:
+                raise TypeError(
+                    "PYTNG ERROR: dtype of array {} does not match TNG dtype int64_t".format(dtype))
+            _int64_t_view = <int64_t[:n_vals] > (< int64_t*> values)
+            data[:, :] = np.asarray(_int64_t_view, dtype=np.int64).reshape(
+                n_atoms, n_values_per_frame)
+
+        elif datatype == TNG_DOUBLE_DATA:
+            if dtype != np.float64:
+                raise TypeError(
+                    "PYTNG ERROR: dtype of array {} does not match TNG dtype double".format(dtype))
+            _double_view = <double[:n_vals] > (< double*> values)
+            data[:, :] = np.asarray(_double_view, dtype=np.float64).reshape(
+                n_atoms, n_values_per_frame)
+
+        else:
+            raise TypeError("PYTNG ERROR: block datatype not understood")
+
+        return data
+
+    cdef tng_function_status _get_data_current_step(self, int64_t block_id,
+                                                    int64_t step,
+                                                    void ** values,
+                                                    int64_t * n_values_per_frame,
+                                                    int64_t * n_atoms,
+                                                    double * prec,
+                                                    char * datatype,
+                                                    bint debug) nogil:
+        """Gets the frame data off disk and into C level arrays
+
+        Parameters
+        ----------
+        block_id : int64_t
+            block id to read
+        step : int64_t
+            integrator step to read
+        values : void **
+            NULL void pointer to hook the data onto
+        n_values_per_frame : int64_t *
+            set to the number of values per frame for the block
+        n_atoms : int64_t *
+            set to the number of atoms or 1 if particle dependent
+        prec : double *
+            set to the precision of the block
+        datatype : char *
+            set to the datatype of the block
+        debug : bint
+            debug the block read
+
+        Notes
+        -----
+        This function is marked nogil and is called without the GIL so cannot
+        contain python or python exceptions. Instead failure is marked by
+        returning TNG_CRITICAL. Cleanup must then be done by the calling code.
```

Comment:
Is this supposed to be 
```reST
:data:`TNG_CRITICAL`
```
Not clear from the docs how the user can check for the status code.